OlinkAnalyze/R/bridgeable_plot.R

```diff
@@ -0,0 +1,294 @@
+#'                                            overlapping_samples,
+#'                                          df1_project_nr = "Explore HT",
+#'                                          df2_project_nr = "Explore 3072",
+#'                                          reference_project = "Explore HT")
+#'
+#'olinkids <- unique(paste0(data$OlinkID,"_",data$Assay))
+#'results <- bridgeable_plts(data = data,
+#'                           median_counts_threshold = 150,
+#'                           min_count = 10)
+#'names(results) <- olinkids
+#'}
+
+bridgeable_plts <- function(data,
+                            median_counts_threshold = 150,
+                            min_count = 10) {
+
+  if (!requireNamespace("ggpubr", quietly = TRUE)) {
+
+    stop("These plots require the ggpubr package.
+         Please install ggpubr before continuing.
+            install.packages(\"ggpubr\")")
+  }
+
+  out_plts <- list()
+
+  ids <- data |>
+    dplyr::select(OlinkID, .data[["BridgingRecommendation"]]) |> #HT OlinkID
+    dplyr::distinct() |>
+    dplyr::pull(OlinkID)
+
+  # Adjusting the platform and add color green for bridgeable assays
+  data <- data |>
+    dplyr::filter(Count > min_count) |>
+    dplyr::mutate(textcol =
+                    dplyr::if_else(.data[["BridgingRecommendation"]] == "No",
+                                   "#A61F04", "#00559E"))
+
+  platforms <- unique(data$Project)
+
+  if (length(platforms) != 2) {
+    cli::cli_abort(
+      message = paste0("Expect 2 platforms. ", length(platforms), "found.")
+    )
+  }
+
+  # IQR/Range plot ----
+  iqr_range_plt <- function(data = data, id = id) {
+
+    col <- data |>
+      dplyr::filter(OlinkID %in% id) |>
+      dplyr::pull(.data[["textcol"]]) |>
+      unique()
+
+    iqr_range_plt <- data |>
+      dplyr::filter(OlinkID %in% id) |>
+      ggplot2::ggplot(ggplot2::aes(x = paste(Assay, OlinkID, sep = "\n"),
+                                   y = NPX, fill = Project)) +
+      ggplot2::geom_violin(alpha = 0.4,
+                           position = ggplot2::position_nudge(x = 0),
+                           width = 0.4) +
+      ggplot2::geom_boxplot(width = .1, outlier.shape = NA, alpha = 0.4) +
+      ggplot2::scale_fill_manual(values = c("#FF1F05", "#00C7E1")) +
+      ggplot2::scale_color_manual(values = c("#FF1F05", "#00C7E1")) +
+      ggplot2::labs(x = "Assay", y = "NPX Distribution", fill = "Platform: ") +
+      ggplot2::guides(fill = ggplot2::guide_legend(nrow = 1, byrow = TRUE)) +
+      OlinkAnalyze::set_plot_theme(font = "") +
+      ggplot2::theme(axis.text.x = ggplot2::element_blank(),
+                     strip.text = ggplot2::element_text(colour = col)) +
+      ggplot2::facet_wrap(. ~ paste(Assay, OlinkID, sep = "\n"),
+                          scales = "free")
+    return(iqr_range_plt)
+  }
+
+  # Correlation plot ----
+  r2_plt <- function(data = data, id = id) {
+
+    # Adding and finding median of negative controls for an approximate LOD
```

Comment:
Remove this line if not incorporating LOD filtering

Reply:
Removed unwanted comments